### PR TITLE
chore: upgrade revm to v3

### PR DESCRIPTION
**Pull request by oblique**

### CODE REVIEW

File: execution/src/evm.rs [hunk 1]
```diff
@@ -174,19 +159,19 @@ impl<'a, R: ExecutionRpc> Evm<'a, R> {
...
-        env.block.coinbase = Address::from_slice(payload.fee_recipient());
-        env.block.timestamp = U256::from(*payload.timestamp());
-        env.block.difficulty = U256::from_little_endian(payload.prev_randao());
+        env.block.number = U256::from(*payload.block_number()).into();
+        env.block.coinbase = Address::from_slice(payload.fee_recipient()).into();
+        env.block.timestamp = U256::from(*payload.timestamp()).into();
+        env.block.difficulty = U256::from_little_endian(payload.prev_randao()).into();
 
-        env.cfg.chain_id = self.chain_id.into();
+        env.cfg.chain_id = self.chain_id.try_into().unwrap();
```
Comment (zvolin) [inline execution/src/evm.rs]: is unwrap safe there?

File: execution/src/evm.rs [hunk 2]
```diff
@@ -220,31 +203,36 @@ impl<'a, R: ExecutionRpc> ProofDB<'a, R> {
...
-        // in WASM environments.
-        //
-        // We are also not sure if this will ever be triggered since batch_fetch_accounts
-        // is pre-fetching the accounts.
-        panic!("not supported in wasm");
+        #[cfg(target_arch = "wasm32")]
+        {
+            let _ = address;
+            let _ = slots;
+            panic!("not supported in wasm");
```
Comment (zvolin) [inline execution/src/evm.rs:226]: if we gonna need it, can we still have `TODO` there?

**Review by zvolin — APPROVED** [review]
nice

File: execution/src/evm.rs (hunk 1 above)
Comment (oblique) [reply execution/src/evm.rs]: yes. it converts a u64 to u256. i can use `expect` is prefer

Comment (oblique) [reply execution/src/evm.rs]: added `expect`

File: execution/src/evm.rs (hunk 2 above)
Comment (oblique) [reply execution/src/evm.rs:226]: added a TODO comment

**Pull request merged**